### ISSUE TITLE
False positive introduced by 1.1.198 associated with PEP 544 

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#2746]
The changes in 1.1.198 flag several patterns involving functions that return `Type[MyProtocol]`. There seems to be "confusion" between a value that has been annotated with `Type[MyProtocol]` versus an actual reference to the `MyProtocol` class object.

For example, consider the following:

```python
from typing import Protocol, Type

class A(Protocol):
    pass

def f() -> Type[A]:
    raise NotImplemented()

def g(x: Type[A]) -> int:
    return 1

y = g(f())  # error: "A" is not a concrete class type and cannot be assigned to type "Type[A]
```

In [hydra-zen](https://github.com/mit-ll-responsible-ai/hydra-zen) I dynamically-generate dataclass types (via [make_dataclass](https://docs.python.org/3/library/dataclasses.html#dataclasses.make_dataclass)) whose behaviors are described by various protocols, so the above issue rears its head under some diverse circumstances -- my pyright tests lit up like a Christmas tree after 1.1.198 was released 😅 . It might be helpful for me to include a couple of these (in simplified form) to help illuminate potential edge cases.

Some boilerplate code:
```python

T1 = TypeVar("T1", covariant=True)
T2 = TypeVar("T2")

class DataClass(Protocol):
    __dataclass_fields__: Dict[str, Field]

class Builds(DataClass, Protocol[T1]): 
    _target_: str
```

I am unable to cast to `Type[Builds]]`:

```python
def builds(x: T2) -> Type[Builds[T2]]:
    out = make_dataclass(<...>)
    return cast(Type[Builds[Importable]], out)  # error: "Builds[T2@builds]" is not a concrete class type and cannot be assigned to type "Type[Builds[T2@builds]]"
```

And here is another particular case that cropped up for me. It is ultimately the exact same issue, I believe, but I figured I'd include it.

```python
# error: Builds[Type[T2@_cast_via_tuple]]" is not a concrete class type and cannot be assigned to type "Type[Builds[Type[T2@_cast_via_tuple]]]
def cast_via_tuple(dest_type: Type[T2]) -> Callable[[T2], Type[Builds[Type[T2]]]]:
    def converter(value):
        return builds(dest_type, tuple(value))

    return converter
```

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.199, which I just published. It will also be included in the next release of pylance.

**Comment by rsokl** [comment]
1.1.199 is working great! I am no longer seeing any false-positives, and it ended up catching some inaccurate/incomplete types that had been lurking in hydra-zen.
